connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -177,8 +182,17 @@ public Response resumeConnector(@PathParam("connector") String connector) {
                                          final @QueryParam("forward") Boolean forward) throws Throwable {
         FutureCallback<List<TaskInfo>> cb = new FutureCallback<>();
         herder.taskConfigs(connector, cb);
-        return completeOrForwardRequest(cb, "/connectors/" + connector + "/tasks", "GET", null, new TypeReference<List<TaskInfo>>() {
-        }, forward);
+
+
+        List<TaskInfo> taskInfoList = completeOrForwardRequest(cb, "/connectors/" + connector + "/tasks", "GET", null,
+            new TypeReference<List<TaskInfo>>() {}, forward);
+
+        List<TaskInfo> maskedTaskInfoList = new ArrayList<>();
+        for (TaskInfo taskInfo : taskInfoList) {
+            maskedTaskInfoList.add(new TaskInfo(taskInfo.id(), herder.maskCredentials(connector, taskInfo.config())));
```

Comment:
Note that this will not mask any passwords in the task configuration that don't appear in the Connector's ConfigDef. Might want to add a comment.